openmdao/core/problem.py

```diff
@@ -690,10 +695,25 @@ def _check_solvers(self):
                         msg += "currently do not support complex step around it."
                         self._setup_errors.append(msg.format(sub.name))
 
+            parts = group.pathname.split('.')
+            for i in range(len(parts)):
+                # if an ancestor solver iterates, we're ok
+                if has_iter_solver['.'.join(parts[:i])]:
+                    is_iterated_somewhere = True
+                    break
+            else:
+                is_iterated_somewhere = False
+
+            # if we're iterated at this level or somewhere above, then it's
+            # ok if we have cycles or states.
+            if is_iterated_somewhere:
+                continue
+
             if isinstance(group.ln_solver, LinearGaussSeidel) and \
                                      group.ln_solver.options['maxiter'] == 1:
                 # If group has a cycle and lings can't iterate, that's
-                # an error.
+                # an error if current lin solver or ancestor lin solver doesn't
+                # iterate.
```

Comment:
good catch.  Turns out I have a test_layout.py file on my machine that I created months ago but didn't add it to the git repo.